examples/batched-solver-from-files/batched-solver-from-files.cpp

```diff
@@ -0,0 +1,223 @@
+#include <string>
+
+
+// @sect3{Type aliases for convenience}
+// Use some shortcuts. In Ginkgo, vectors are seen as a gko::matrix::Dense
+// with one column/one row. The advantage of this concept is that using
+// multiple vectors is a now a natural extension of adding columns/rows are
+// necessary.
+using value_type = double;
+using real_type = gko::remove_complex<value_type>;
+using index_type = int;
+using size_type = gko::size_type;
+using vec_type = gko::matrix::BatchDense<value_type>;
+using real_vec_type = gko::matrix::BatchDense<real_type>;
+using mtx_type = gko::matrix::BatchCsr<value_type, index_type>;
+using solver_type = gko::solver::BatchBicgstab<value_type>;
+
+
+int main(int argc, char* argv[])
+{
+    // Print the ginkgo version information.
+    std::cout << gko::version_info::get() << std::endl;
+
+    if (argc == 2 && (std::string(argv[1]) == "--help")) {
+        std::cerr << "Usage: " << argv[0] << " [executor] " << std::endl;
+        std::exit(-1);
+    }
+
+    // @sect3{Where do you want to run your solver ?}
+    // The gko::Executor class is one of the cornerstones of Ginkgo. Currently,
+    // we have support for an gko::OmpExecutor, which uses OpenMP
+    // multi-threading in most of its kernels,
+    // a gko::ReferenceExecutor, a single threaded specialization of
+    // the OpenMP executor and a gko::CudaExecutor which runs the code on a
+    // NVIDIA GPU if available.
+    // @note With the help of C++, you see that you only ever need to change the
+    // executor and all the other functions/ routines within Ginkgo should
+    // automatically work and run on the executor with any other changes.
+    const auto executor_string = argc >= 2 ? argv[1] : "reference";
+    std::map<std::string, std::function<std::shared_ptr<gko::Executor>()>>
+        exec_map{
+            {"omp", [] { return gko::OmpExecutor::create(); }},
+            {"cuda",
+             [] {
+                 return gko::CudaExecutor::create(0, gko::OmpExecutor::create(),
+                                                  true);
+             }},
+            {"hip",
+             [] {
+                 return gko::HipExecutor::create(0, gko::OmpExecutor::create(),
+                                                 true);
+             }},
+            {"dpcpp",
+             [] {
+                 return gko::DpcppExecutor::create(0,
+                                                   gko::OmpExecutor::create());
+             }},
+            {"reference", [] { return gko::ReferenceExecutor::create(); }}};
+
+    // executor where Ginkgo will perform the computation
+    const auto exec = exec_map.at(executor_string)();  // throws if not valid
+
+    const std::string problem_descr_str = argc >= 3 ? argv[2] : "pores_1";
+    const size_type num_systems = argc >= 4 ? std::atoi(argv[3]) : 2;
+    const size_type num_duplications = argc >= 5 ? std::atoi(argv[4]) : 2;
+    const std::string batch_scaling = argc >= 6 ? argv[5] : "none";
```

Comment:
I think it might make sense to describe these with a comment since they are tied to where you find the matrices, although it's somewhat easy to understand when looking few lines below.

Reply:
Another way to do it is to put this all together in the comment of `Read batch from files`.